src/main/java/io/github/syst3ms/skriptparser/lang/properties/PropertyConditional.java

```diff
@@ -0,0 +1,136 @@
+    private Expression<P> performer;
+    private String performerName;
+    private ConditionalType conditionalType;
+    private String propertyName;
+
+    public Expression<P> getPerformer() {
+        return performer;
+    }
+
+    public void setPerformer(Expression<P> performer) {
+        this.performer = performer;
+    }
+
+    /**
+     * This default {@code init()} implementation automatically properly sets the performer in this condition,
+     * which can be accessed using {@link #getPerformer()}. If this implementation is overridden for one reason
+     * or another, it must call {@link #setPerformer(Expression)} properly.
+     * @param expressions an array of expressions representing all the expressions that are being passed
+     *                    to this syntax element.
+     * @param matchedPattern the index of the pattern that was successfully matched. It corresponds to the order of
+     *                       the syntaxes in registration
+     * @param parseContext an object containing additional information about the parsing of this syntax element, like
+     *                    regex matches and parse marks
+     * @return whether the initialization was successful or not.
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+        setPerformer((Expression<P>) expressions[0]);
+        setNegated(matchedPattern == 1);
+        return true;
+    }
+
+    @Override
+    public boolean check(TriggerContext ctx) {
+        return check(ctx, performer.getValues(ctx));
+    }
+
+    public abstract boolean check(TriggerContext ctx, P[] performers);
+
+    @Override
+    public String toString(@Nullable TriggerContext ctx, boolean debug) {
+        return toString(ctx, debug, performer, conditionalType, propertyName);
+    }
+
+    private String toString(@Nullable TriggerContext ctx, boolean debug,
+                            Expression<P> perf,
+                            ConditionalType conditionalType,
+                            String property) {
+        switch (conditionalType) {
+            case BE:
+                return perf.toString(ctx, debug) + (perf.isSingle() ? " is " : " are ") + (isNegated() ? "not " : "") + property;
+            case CAN:
+                return perf.toString(ctx, debug) + (isNegated() ? " can't " : " can ") + property;
+            case HAVE:
+                if (perf.isSingle())
+                    return perf.toString(ctx, debug) + (isNegated() ? " doesn't have " : " has ") + property;
+                else
+                    return perf.toString(ctx, debug) + (isNegated() ? " don't have " : " have ") + property;
+            default:
+                throw new AssertionError();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void register(SkriptRegistration reg, Object... args) {
```

Comment:
Please don't use a spread of Object here, it's clearly not readable when we try to read args of the method from another file.

Reply:
I do not understand this. In my opinion, changing it to an array would defeat the readability more than using the spread identifier, or am I missing some point?

Reply:
When you use a spread object like ``Object... args`` and use it as Object array, when using the method, we don't know what args are. In this case, you should not use that.

Reply:
That's the whole point, isn't it? The developer can give as many arguments of the desired type as he wants. If he uses the wrong one though, then it's up to the one making the SelfRegistrable class to notice that and output an error. I can't make this parameter of a certain type since we actually have no control over what the input is. We can only check if it's correct and then use that information to register the class.

Reply:
This is exactly what's done in SkriptLambda. If you want an unknown number of arguments of unknown types, use an Object array. And then once you have that you might as well make it a vararg.

Reply:
So why below we have a cast?
```
performerName = (String) args[0];
conditionalType = (ConditionalType) args[1];
propertyName = (String) args[2];
```
It doesn't make sense to use an array of object as parameter when just after that you cast all values of the array.

Reply:
Because the SelfRegistrable (should be SelfRegisterable) interface isn't specific to PropertyCondition. If you instead use it for a PropertyExpression, an Effect or a CodeSection, the arguments wouldn't be the same, and there isn't a better way to allow for all these different formats of arguments at once than an Object vararg.

Reply:
I looked for a bit and saw that Registrable is correct as an adjective (in the same way Registrar is correct).

Reply:
It's just my bad, nevermind, forget my review, sorry about that

Reply:
No problem! It's better to have a discussion about it than none.